app/src/main/java/com/edulectronics/tinycircuit/Models/Components/Connectors/ConnectionPoint.java

```diff
@@ -0,0 +1,80 @@
+package com.edulectronics.tinycircuit.Models.Components.Connectors;
+
+import com.edulectronics.tinycircuit.Models.Components.Component;
+import com.edulectronics.tinycircuit.Models.Components.Powersource;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by Maaike on 28-11-2016.
+ */
+
+public class ConnectionPoint {
+
+    public ConnectionPointOrientation orientation;
+    private Component parentComponent;
+    private List<ConnectionPoint> connections = new ArrayList<ConnectionPoint>();
+
+    /*
+    TODO: Depending on implementation for current calulations, voltageIn and -Out might
+    be replaced by a single Voltage value.
+    */
+    private double voltageOut;
+    private double voltageIn;
+
+    public ConnectionPoint(Component parent, ConnectionPointOrientation orientation) {
+        this.parentComponent = parent;
+        this.orientation = orientation;
+    }
+
+    public double getVolgateOut() {
+        return this.voltageOut;
+    }
+
+    // Set a new output voltage. Only a component is allowed to do this to its own connection
+    // points.
+    public void setVoltageOut(double voltage) {
+        this.voltageOut = voltage;
+        for (ConnectionPoint connectionPoint: connections) {
+            connectionPoint.calculateVoltageIn();
+        }
+    }
+
+    // Returns the accumulated voltage of all connectionpoints.
+    private void calculateVoltageIn() {
+        double voltage = 0;
+        for (ConnectionPoint connectionPoint: connections) {
```

Comment:
Ik weet niet of java 7 het al toe laat of het pas in java 8 is gekomen, maar als het kan liever iets als
`
voltage = 
       myList.stream()
              .mapToInt(connectionPoint -> connectionPoint.getVoltageOut())
              .sum();
`